lib/streamlit/type_util.py

```diff
@@ -233,6 +235,18 @@ def is_dataframe_like(obj: object) -> TypeGuard[DataFrameLike]:
     return any(is_type(obj, t) for t in _DATAFRAME_LIKE_TYPES)
 
 
+def is_snowpark_dataframe(obj: object) -> bool:
```

Comment:
Looks good - can you add a test for this function? It should cover all the clauses here: passing a snowpark dataframe, passing a list of snowpark rows, passing a list of some unrelated object, passing an empty list, and passing an arbitrary non-list object.

Reply:
Also could use a short docstring explaining the circumstances that it will return true for.

Reply:
@tconkling thank you for the review, the tests are present now. I've created `tests.snowflake.snowpark` module as a mock for Snowpark, because Snowpark supports only Python 3.8, and our CI tests range from min python version to max python version, is that ok?